lesson_10/test_homework_10.py

```diff
@@ -7,15 +7,122 @@
 На оцінку впливає як якість тестів так і розмір тестового покриття. 
 Мінімум на 10 балів - 1 правильно задизайнений позитивний тест на функцію.
 """
-from homeworks import func
+
+
 import unittest
+from homeworks import add_numbers, reverse_string, long_word, find_substring, new_average
+
+class TestHomeworkFunctions(unittest.TestCase):
+
+    def test_add_numbers_1(self):
+        """Tests for the add_numbers function."""
+        actual = add_numbers(3, 2)
+        expected = 5
+        self.assertEqual(actual, expected, f"Expected {expected}, but got {actual}")
+        
+    def test_add_numbers_2(self):
+        actual = add_numbers(-1, 1)
+        expected = 0
+        self.assertEqual(actual, expected, f"Expected {expected}, but got {actual}")
+        
+    def test_add_numbers_3(self):
+        actual = add_numbers(1.5, 2.5)
+        expected = 4.0
+        self.assertEqual(actual, expected, f"Expected {expected}, but got {actual}")
+
+    def test_add_numbers_4(self):
+        with self.assertRaises(TypeError):
+            add_numbers("3", 2)  
+
+
+    def test_new_average_1(self):
+        """Tests for the new_average function."""
+        actual = new_average([1, 2, 3, 4, 5])
+        expected = 3.0
+        self.assertEqual(actual, expected, f"Expected {expected}, but got {actual}")
+        
+    def test_new_average_2(self):
+        actual = new_average([10, 20, 30])
+        expected = 20.0
+        self.assertEqual(actual, expected, f"Expected {expected}, but got {actual}")
+
+    def test_new_average_3(self):
+        actual = new_average([2])
+        expected = 2.0
+        self.assertEqual(actual, expected, f"Expected {expected}, but got {actual}")
+        
+    def test_new_average_4(self):
+        with self.assertRaises(ValueError):
+            new_average([])        
+
+
+    def test_reverse_string_1(self):
+        """Tests for the reverse_string function."""
+        actual = reverse_string("hello")
+        expected = "olleh"
+        self.assertEqual(actual, expected, f"Expected {expected}, but got {actual}")
+   
+    def test_reverse_string_2(self):
+        actual = reverse_string("Python")
+        expected = True  
+        self.assertTrue(actual == "nohtyP", f"Expected {expected}, but got {actual}")
```

Comment:
```suggestion
        self.assertTrue(actual == "nohtyP", f"Expected {expected}, but got {actual}")
```
ну по тут явний  assertEqual, це годиться лише на погратися